src/custom-element.js

```diff
@@ -495,22 +516,18 @@ function createBaseCustomElementClass(win) {
      * @param {boolean} onLayout Whether this was called after a layout.
      */
     preconnect(onLayout) {
+      devAssert(this.isUpgraded());
       if (onLayout) {
-        this.implementation_.preconnectCallback(onLayout);
+        this.impl_.preconnectCallback(onLayout);
       } else {
         // If we do early preconnects we delay them a bit. This is kind of
         // an unfortunate trade off, but it seems faster, because the DOM
         // operations themselves are not free and might delay
         startupChunk(this.getAmpDoc(), () => {
-          const TAG = this.tagName;
-          if (!this.ownerDocument) {
-            dev().error(TAG, 'preconnect without ownerDocument');
```

Comment:
FMI: why is this dev error no longer necessary?

Reply:
It was initial fact-finding check. I was (long ago) researching how often elements end-up in a disconnected state by this time. It's no longer valuable.